### PR TITLE
adding tags definition for IAM Role

### PR DESCRIPTION
**Pull request by Rohit1509**
Signed-off-by: Rohit Joshi <rohit.prasad.joshi@sap.com>

### Description

Adding tags definition in IAM role resource

### Issues Resolved

N/A

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [ ] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [ ] All Unit Tests pass
- [ ] `rake lint` passes
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_role.rb [hunk 1]
```diff
@@ -48,6 +48,15 @@ def inline_policies
     end
   end
 
+  def tags
+    begin
+      tag_list = @aws.iam_client.get_role(role_name: opts[:role_name]).role.tags
```
Comment (sathish-progress) [inline libraries/aws_iam_role.rb:53]: please wrap with exception handler `catch_aws_errors`

Comment (Rohit1509) [reply libraries/aws_iam_role.rb]: @sathish-progress , function definition updated

File: libraries/aws_iam_role.rb [hunk 2]
```diff
@@ -48,6 +48,18 @@ def inline_policies
     end
   end
 
+  def tags
+    tag_list = []
+    catch_aws_errors do
+      begin
+        tag_list = @aws.iam_client.get_role(role_name: opts[:role_name]).role.tags
+      rescue
```
Comment (sathish-progress) [inline libraries/aws_iam_role.rb:56]: there is no point having both `catch_aws_errors` and `begin; rescue`!
could you remove `begin; rescue` block from here ?

Comment (Rohit1509) [reply libraries/aws_iam_role.rb]: @sathish-progress its updated

**Review by sathish-progress — APPROVED** [review]

**Comment by Rohit1509** [comment]
@sathish-progress can this be merged please ? This is a high priority item for us. Thanks!

**Pull request merged**